thermocouples_reference/function_types.py

```diff
@@ -100,7 +100,7 @@ def __call__(self,T,derivative=0,out_of_range="raise"):
         if out_of_range not in ["raise", "nan", "extrapolate"]:
             raise ValueError("invalid out_of_range parameter",out_of_range)
 
-        T = np.array(T, copy=False, order='A')
+        T = np.array(T, order='A')
```

Comment:
Why does this change fix the issue?

Reply:
They changed the meaning of False, as described [here](https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword). It now throws an exception if a copy is needed, which is the case here.